engine/src/cmd/damageable.cpp

```diff
@@ -247,15 +247,26 @@ void Damageable::ApplyDamage( const Vector &pnt,
         //static float hull_dam_to_eject    =
         //    GameConfig::GetVariable( "physics", "hull_damage_to_eject", 100.0 );
         static float auto_eject_percent =
-                GameConfig::GetVariable( "physics", "autoeject_percent", 0.5 );
+                GameConfig::GetVariable( "physics", "autoeject_percent", 0.5f );
         static bool player_autoeject =
                 GameConfig::GetVariable( "physics", "player_autoeject", true );
 
-        if(shot_at_is_player && player_autoeject) {
-            unit->EjectCargo( (unsigned int) -1 );
-        } else if(rand() < (RAND_MAX*auto_eject_percent) && unit->isUnit() == _UnitType::unit &&
-                  unit->faction != neutralfac && unit->faction != upgradesfac) {
-            unit->EjectCargo( (unsigned int) -1 );
+        if (shot_at_is_player) {
+            if (player_autoeject
+                    && rand() < (RAND_MAX * auto_eject_percent)) {
```

Comment:
This would be better as a function of some sort, especially since you use it twice right here - or calculate it ahead of time and store in a variable